### PR TITLE
[DOC] Add Ingress doc

### PR DESCRIPTION
**Pull request by svekars**

### CODE REVIEW

**Comment by svekars** [comment]
@pappasilenus please review when you have a moment.

File: doc/docs/master/deploy-manage/manage/configure-external-access.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+cloud environments as well, but a production environment might require you to
+define more sophisticated inbound connection rules.
+
+Kubernetes provides multiple ways to deliver external traffic to a service,
+including the following:
+
+* Through a Service with `type: NodePort`. A NodePort service provides basic
+access to services. By default, the `pachd` service is deployed as `NodePort`
+to simplify interaction with your localhost. `NodePort` is a limited solution
+that might not be reliable in production deployments.
```
Comment (pappasilenus) [inline doc/docs/master/deploy-manage/manage/configure-external-access.md:18]: change to 
```
that might not be considered reliable or secure enough for production deployments.
````

File: doc/docs/master/deploy-manage/manage/configure-external-access.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  Exposing your application to an outside world might pose a security
+  risk to your data and organization. Make sure that you have Transport
+  Layer Security (TLS) enabled for Ingress connections.
+
+* **Use Pachyderm authentication**
+
+  Pachyderm authentication must be enabled and access provided to a
+  verified list of users. Pachyderm authentication is an additional
+  security layer to protect your data from malicious attacks.
+  If you cannot use Pachyderm authentication providers, you need to
```
Comment (pappasilenus) [inline doc/docs/master/deploy-manage/manage/configure-external-access.md:57]: Change to 

```
If you cannot use Pachyderm authentications providers, it is highly recommended, for security reasons, that you
```

File: doc/docs/master/deploy-manage/manage/configure-external-access.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  use Pachyderm port-forwarding. Exposing Pachyderm services through
+  an ingress without Pachyderm authentication might result in your
+  data being compromised.
```
Comment (pappasilenus) [inline doc/docs/master/deploy-manage/manage/configure-external-access.md:60]: change `your data being compromised` to 

```
your Pachyderm and Kubernetes clusters being compromised, along with your data.
```

File: doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+   ```bash
+   kubectl apply -f https://raw.githubusercontent.com/pachyderm/pachyderm/master/examples/traefik-ingress/roles.yaml
+   kubectl apply -f https://raw.githubusercontent.com/pachyderm/pachyderm/master/examples/traefik-ingress/traefik-daemonset.yaml
+   ```
+
+1. Deploy the Ingress resource. The following text is an example
+metadata file. You need to adjust it to your deployment:
+
+   <script src="https://gist.github.com/svekars/9c665571b665bbd999e050cb829adf2e.js"></script>
```
Comment (pappasilenus) [inline doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md:47]: I can't see this gist. even when I view file on github.  is it only viewable on our docs site?  Be nice to solve that.

I saved the gist embed to a local file and viewed it, and it looks fine and correct!

File: doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+   * `port` — the ports on which the `grpc-proxy` and the `dash`
+   application will run. It can be the same or different ports.
+
+   When done editing, save the file and run:
+
+   ```bash
+   kubectl apply -f traefik-ingress-minikube.yaml
+   ```
+
+   Make sure that you modify the following parameters:
```
Comment (pappasilenus) [inline doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md:62]: I think likes 62-67 repeat lines 49-54?

File: doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+the following steps:
+
+1. Deploy a Pachyderm cluster to a cloud provider of your choice as
+described in [Deploy Pachyderm](../../deploy/).
+For production deployments, deploy with the `--tls` flag.
+1. Enable authentication as described in [Configure Access Controls](../../../enterprise/auth/auth/).
+1. Deploy the traefik ingress controller:
+
+   ```bash
+   kubectl apply -f https://raw.githubusercontent.com/pachyderm/pachyderm/master/examples/traefik-ingress/roles.yaml
```
Comment (pappasilenus) [inline doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md:40]: I think we should point them to traefik's deployment documentation here, instead?

https://docs.traefik.io/v1.7/user-guide/kubernetes/

I'm worried about us having out-of-date traefik deployment resources here.

File: doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+port-forwarding, you might want to expose the Pachyderm UI, `dash`
+through an Ingress resource. To do so, you need to deploy an ingress
+controller and ingress resource object on the Kubernetes cluster that
+runs your Pachyderm cluster. While Kubernetes supports multiple ingress
+controllers, not all of them might work seamlessly with Pachyderm.
+See [Pachyderm Ingress Requirements](../configure-external-access/#pachyderm-ingress-requirements/) for more information.
+
+In addition, follow these recommendations:
+
+* `DaemonSets` are generally more preferred than `Deployments`.
```
Comment (pappasilenus) [inline doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md:16]: Let's replace line 16 with

```
Follow the recommendations in [the Traefik Kubernetes installation documentation](https://docs.traefik.io/v1.7/user-guide/kubernetes/#deploy-traefik-using-a-deployment-or-daemonset) 
that are appropriate for your situation for installing Traefik.
```

File: doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md (hunk 6 above)
Comment (svekars) [reply doc/docs/master/deploy-manage/manage/expose-pach-ui-ingress.md:45]: added above.

**Pull request merged**